web/pgadmin/tools/sqleditor/static/js/components/sections/ResultSetToolbar.jsx

```diff
@@ -218,7 +236,7 @@ export function ResultSetToolbar({query,canEdit, totalRowCount}) {
           <PgIconButton title={gettext('Copy options')} icon={<KeyboardArrowDownIcon />} splitButton
             name="menu-copyheader" ref={copyMenuRef} onClick={openMenu} />
           <PgIconButton title={gettext('Paste')} icon={<PasteIcon />}
-            shortcut={queryToolPref.btn_paste_row} disabled={!canEdit} onClick={pasteRows} />
+            shortcut={queryToolPref.btn_paste_row} disabled={buttonsDisabled['paste-rows']} onClick={pasteRows} />
```

Comment:
do not disable paste rows, it can come from any source outside pgadmin.

Reply:
Done

Reply:
Done